src/main/java/com/powsybl/openloadflow/sensi/DcSensitivityAnalysis.java

```diff
@@ -608,20 +756,62 @@ protected DenseMatrix initContingencyRhs(LfNetwork lfNetwork, EquationSystem equ
                                         EquationSystem equationSystem, Collection<Contingency> nonLosingConnectivityContingencies,
-                                        Map<List<Set<ComputedContingencyElement>>, List<Contingency>> contingenciesByGroupOfElementsBreakingConnectivity) {
+                                        Map<Set<ComputedContingencyElement>, List<Contingency>> contingenciesByGroupOfElementsBreakingConnectivity) {
         for (Contingency contingency : contingencies) {
-            List<Set<ComputedContingencyElement>> groupOfElementsBreakingConnectivity = getGroupOfElementsBreakingConnectivity(lfNetwork, states,
+            Set<ComputedContingencyElement> groupOfElementsBreakingConnectivity = getGroupOfElementsBreakingConnectivity(lfNetwork, states,
                     contingency.getElements().stream().map(element -> contingenciesElements.get(element.getId())).collect(Collectors.toList()), equationSystem);
             if (groupOfElementsBreakingConnectivity.isEmpty()) { // connectivity not broken
                 nonLosingConnectivityContingencies.add(contingency);
-            } else if (groupOfElementsBreakingConnectivity.stream().mapToInt(Set::size).max().getAsInt() > 1) {
-                throw new PowsyblException("The contingency " + contingency.getId() + " breaks the connectivity on more than one branch.");
             } else {
                 contingenciesByGroupOfElementsBreakingConnectivity.computeIfAbsent(groupOfElementsBreakingConnectivity, key -> new LinkedList<>()).add(contingency);
             }
         }
     }
 
+    private Set<String> getElementsToReconnect(GraphDecrementalConnectivity<LfBus> connectivity, Set<ComputedContingencyElement> breakingConnectivityCandidates) {
+        Set<String> elementsToReconnect = new HashSet<>();
+
+        Map<Pair<Integer, Integer>, Set<ComputedContingencyElement>> elementsByConnectedComponents =
+                breakingConnectivityCandidates.stream().collect(Collectors.toMap(
+                    element -> Pair.of(connectivity.getComponentNumber(element.getLfBranch().getBus1()), connectivity.getComponentNumber(element.getLfBranch().getBus2())),
+                    element -> new HashSet<>(Collections.singletonList(element)),
+                    (existing, replacement) -> {
+                        existing.addAll(replacement);
+                        return existing;
+                    }
+                ));
+
+        Map<Integer, Set<Integer>> connections = new HashMap<>();
+        connections.put(0, Collections.singleton(0));
+        for (Set<LfBus> smallComponent : connectivity.getSmallComponents()) {
+            int ccNumber = connectivity.getComponentNumber(smallComponent.iterator().next());
+            connections.put(ccNumber, new HashSet<>(Collections.singletonList(ccNumber)));
+        }
```

Comment:
It would be clearer to replace the loop with a `for (int i=0; i<nbComponents; i++) connections.put(i, Collections.singleton(i));`

Reply:
Yes, indeed, I wasn't sure that the connectivity guaranteed the component number to always stay between 0 and the number of components.
But reading at the documentation in the interface, this is the case, so I will change this